### PR TITLE
Fix class docstring for tracking datasets

### LINKED ISSUE #1823: tracking.JSONDataSet documentation and docstring are not accurate
Here is the source code for `tracking.JSONDataSet`: https://kedro.readthedocs.io/en/latest/_modules/kedro/extras/datasets/tracking/json_dataset.html#JSONDataSet

The `_load` method has been overriden to raise a `DataSetError` when called.

The documentation does not reflect that well. What has to be fixed:

- The `tracking.JSONDataSet` documentation and API ref should mention that this dataset cannot be loaded (see https://kedro.readthedocs.io/en/latest/kedro.extras.datasets.tracking.JSONDataSet.html)
- The `tracking.JSONDataSet` class docstring should not call `data_set.load()` (this code snippet that was probably copied from the `json.JSONDataSet`)

### PR DESCRIPTION
**Pull request by clotildeguinard**
## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

Resolves #1823 

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1838"></a>

### CODE REVIEW

**Comment by MerelTheisenQB** [comment]
Thank you so much for these updates @clotildeguinard ⭐  Would you mind updating the [`tracking.MetricsDataSet`](https://github.com/kedro-org/kedro/blob/main/kedro/extras/datasets/tracking/metrics_dataset.py) in the same way?

Also the DCO check is currently failing, we have [instructions in our docs on how to make it pass](https://kedro.readthedocs.io/en/latest/contribution/developer_contributor_guidelines.html?highlight=dco#developer-certificate-of-origin). Let me know if you have any trouble with it.

**Comment by clotildeguinard** [comment]
Ok @MerelTheisenQB  I have setup DCO, please let me know if there is anything else to improve in the PR as this is my first contribution!

**Review by MerelTheisenQB — APPROVED** [review]
Thanks a lot for this contribution 😄 ⭐

**Comment by MerelTheisenQB** [comment]
> Ok @MerelTheisenQB I have setup DCO, please let me know if there is anything else to improve in the PR as this is my first contribution!

It looks great! Thank you and congrats on your first Kedro contribution 🥳

File: kedro/extras/datasets/tracking/json_dataset.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class JSONDataSet(JDS):
     """``JSONDataSet`` saves data to a JSON file using an underlying
     filesystem (e.g.: local, S3, GCS). It uses native json to handle the JSON file.
     The ``JSONDataSet`` is part of Kedro Experiment Tracking.
-    The dataset is versioned by default.
+    The dataset is read-only and it is versioned by default.
```
[the remark this answers is not in the document]
Comment (MerelTheisenQB) [reply kedro/extras/datasets/tracking/json_dataset.py:15]: Oh gosh good catch!! 😅 It should indeed by `write-only`

[the remark this answers is not in the document]
Comment (noklam) [reply kedro/extras/datasets/tracking/json_dataset.py:15]: Cool, I will just go ahead and change it, hope this is ok since this is quite straight forward.

**Review by noklam — APPROVED** [review]

**Comment by clotildeguinard** [comment]
@noklam thanks a lot for spotting and fixing the "read-only" mistake :facepalm:

**Comment by noklam** [comment]
@clotildeguinard congrats on your first Kedro PR😁

**Pull request merged**